src/BrazeMessage.tsx

```diff
@@ -19,13 +22,19 @@ const COMPONENT_MAPPINGS: ComponentMapping = {
 };
 
 export type Props = {
-    onButtonClick: (buttonIndex: number) => void;
+    logButtonClickWithBraze: BrazeClickHandler;
+    submitComponentEvent: (componentEvent: OphanComponentEvent) => void;
     componentName: string;
     brazeMessageProps: BrazeMessageProps;
 };
 
 export const buildBrazeMessageComponent = (mappings: ComponentMapping): React.FC<Props> => {
-    const BrazeMessageComponent = ({ onButtonClick, componentName, brazeMessageProps }: Props) => {
+    const BrazeMessageComponent = ({
+        logButtonClickWithBraze,
+        submitComponentEvent,
+        componentName,
```

Comment:
I wonder if this `CommonComponentProps` vs `Props` distinction could be simplified away, by treating the `componentName` as a separate argument rather than a property. This is totally off topic for the PR though so let's chat separately. It has come up because these types made me go a bit cross-eyed while trying to work out what's happening here.